### PR TITLE
SPCTR-2397: Fix pagination bug in Post Grid causing unintended changes to second grid's pagination.

### PR DESCRIPTION
**Pull request by chandan-30**
### Description
This PR addresses a bug where changing the pagination settings for one Post Grid would also change the pagination settings for a second Post Grid on the same page. To fix we added a click event listener to the pagination buttons. Upon clicking a pagination button, an AJAX call is sent to retrieve the relevant page data. Once the response is received, we update the Post Grid with the new markup, replacing the old markup with the new one.

This approach ensures that each Post Grid maintains its own pagination state and that changes to one grid's pagination settings do not affect any other grids on the page.

### Screenshots
https://d.pr/v/l2h1oY

### Checklist:
- [x] My code is tested
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: blocks-config/post/class-uagb-post.php [hunk 1]
```diff
@@ -1181,6 +1183,22 @@ public function post_grid_callback( $attributes ) {
 			return ob_get_clean();
 		}
 
+		/**
+		 * Renders the post grid block on pagination clicks.
+		 *
+		 * @since 2.4.1
```
Comment (AbhijitNage123) [inline blocks-config/post/class-uagb-post.php:1189]: New function should have @since x.x.x

File: blocks-config/post/class-uagb-post.php [hunk 2]
```diff
@@ -1181,6 +1183,22 @@ public function post_grid_callback( $attributes ) {
...
+		 * 
+		 * @return void
+		 */
+		public function post_grid_pagination_ajax_callback() {
+			check_ajax_referer( 'uagb_grid_ajax_nonce', 'nonce' );
+			$attr = isset( $_POST['attr'] ) ? json_decode( stripslashes( $_POST['attr'] ), true ) : array(); //phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
```
Comment (AbhijitNage123) [inline blocks-config/post/class-uagb-post.php:1195]: check  isset( $_POST['attributes'] ) and wp_send_json_success, wp_send_json_error inside function. Refer - https://github.com/brainstormforce/ultimate-addons-for-gutenberg/blob/450ce0a25b98f3a96d0425bebe768bf024492da3/blocks-config/post/class-uagb-post.php#L1509

File: blocks-config/post/class-uagb-post.php (hunk 1 above)
Comment (chandan-30) [reply blocks-config/post/class-uagb-post.php]: Made changes. Please re-review.

File: blocks-config/post/class-uagb-post.php (hunk 2 above)
Comment (chandan-30) [reply blocks-config/post/class-uagb-post.php]: Made changes. Please re-review.

**Review by AbhijitNage123 — APPROVED** [review]

**Review by RajkiranBagal — APPROVED** [review]

**Pull request merged**